gladier/managers/login_manager.py

```diff
@@ -0,0 +1,102 @@
+        and the current token we now have is invalid and can no longer be used. The solution is
+        a new login to get a new token.
+        """
+        log.debug(f'Tracking scope change: {scopes}')
+        self.scope_changes = self.scope_changes | set(scopes)
+
+    def get_authorizers(self) -> List[Union[AccessTokenAuthorizer, RefreshTokenAuthorizer]]:
+        self.ensure_logged_in()
+        return self.get_native_client().get_authorizers_by_scope()
+
+    def ensure_logged_in(self):
+        missing = self.missing_authorizers
+        if missing:
+            log.debug(f'Missing the following scopes requiring a login: {missing}')
+            kwargs = {}
+            if self.scope_changes:
+                kwargs['force'] = True
+            self.login(requested_scopes=missing, **kwargs)
+
+    def login(self, **login_kwargs):
+        """Login to the Gladier client. This will ensure the user has the correct
+        tokens configured but it DOES NOT guarantee they are in the correct group to
+        run a flow. Can be run both locally and on a server.
+        See help(fair_research_login.NativeClient.login) for a full list of kwargs.
+        """
+        nc = self.get_native_client()
+        # if self.is_logged_in() and login_kwargs.get('force') is not True:
+        #     log.debug('Already logged in, skipping login.')
+        #     return
+        log.info('Initiating Native App Login...')
+        log.debug(f'Requesting Scopes: {self.required_scopes}')
+        login_kwargs['requested_scopes'] = login_kwargs.get('requested_scopes',
+                                                            self.required_scopes)
+        login_kwargs['refresh_tokens'] = login_kwargs.get('refresh_tokens', True)
+        nc.login(**login_kwargs)
+        self.authorizers = nc.get_authorizers_by_scope()
+
+    def logout(self):
+        """Log out and revoke this client's tokens. This object will no longer
+        be usable until a new login is called.
+        """
+        log.info(f'Revoking the following scopes: {self.scopes}')
+        self.get_native_client().logout()
+        # Clear authorizers cache
+        self.authorizers = dict()
+
+    def get_native_client(self):
+        """
+        fair_research_login.NativeClient is used when ``authorizers`` are not provided to __init__.
+        This enables local login to the Globus Automate Client, FuncX, and any other Globus
+        Resource Server.
+
+        :return: an instance of fair_research_login.NativeClient
+        """
+        if getattr(self, 'client_id', None) is None:
+            raise AuthException(
+                'Gladier client must be instantiated with a '
+                '"client_id" to use "login()!'
+            )
+        return fair_research_login.NativeClient(client_id=self.client_id,
+                                                app_name=self.app_name,
+                                                token_storage=self.storage)
```

Comment:
The Login Manager here had some inspiration from lots of places, including the flows client, the CLI, and FuncX. I plan to tidy this up a bit and add a protocol to allow custom login managers.